### PR TITLE
Run molecule lint and converge tests in github actions for aws provider

### PR DESCRIPTION
**Pull request by 14rcole**

### CODE REVIEW

File: .github/workflows/linting.yml [hunk 1]
```diff
@@ -25,3 +25,7 @@ jobs:
         flake8 --exclude=\.eggs,tests,docs,config/Dockerfiles --ignore=E124,E303,W504 --max-line-length 80 .
         #flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics;
         # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+    - name: "Execute linting for AWS role"
+      run: |
```
Comment (samvarankashyap) [inline .github/workflows/linting.yml]: @14rcole All workflows are independent executors. 
you will have to run make sure the container has molecule installed before you run any of the molecule commands.

File: .github/workflows/molecule-tests.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    name: Build on ${{ matrix.container }}
+    runs-on: ubuntu-latest
+    container: ${{ matrix.container }}
+    strategy:
+      matrix:
+        container: ['centos:7', 'centos:8', 'fedora:30', 'fedora:31', 'fedora:32']
+      fail-fast: false
+    steps:
+    - name: "Execute AWS molecule tests"
+      run: |
```
Comment (samvarankashyap) [inline .github/workflows/molecule-tests.yml]: @14rcole Install molecule first from requiements.txt then run molecule command

File: requirements.txt [hunk 3]
```diff
@@ -26,3 +26,4 @@ tqdm==4.36.1
 pyzmq==18.1.1
 mock
 google-auth
+molecule<=2.22
```
Comment (samvarankashyap) [inline requirements.txt]: @14rcole Please move molecule to test dependencies in 
https://github.com/CentOS-PaaS-SIG/linchpin/blob/develop/setup.py#L56

File: .github/workflows/linting.yml (hunk 1 above)
Comment (14rcole) [reply .github/workflows/linting.yml]: Got it, thanks

**Comment by 14rcole** [comment]
@samvarankashyap It looks like the `pip install -e .[tests]` command isn't running

**Comment by samvarankashyap** [comment]
@14rcole 
you have to add the molecule<=2.22 
https://github.com/CentOS-PaaS-SIG/linchpin/blob/1f97fdbc6d9d7ca7e3347ce107d86cab823bdcc7/setup.py#L57 
Instead of the test_require section. 
That's the reason the PR is not installing a molecule as a dependency.

**Comment by samvarankashyap** [comment]
@14rcole Needs rebase, PR has conflicts with setup.py

**Comment by samvarankashyap** [comment]
@14rcole Please create an Issue linked to this PR if not present before you merge this PR

**Comment by 14rcole** [comment]
[merge]

**Pull request merged**